morphic.js

```diff
@@ -7860,6 +7909,41 @@ StringMorph.prototype.endOfLine = function () {
     return this.text.length;
 };
 
+StringMorph.prototype.previousWordFrom = function (aSlot) {
+    // answer the slot (index) slots indicating the position of the
+    // previous word to the left of aSlot
+    var index = aSlot - 1;
+    
+    // while the current character is non-word one, we skip it, so that
+    // if we are in the middle of a non-alphanumeric sequence, we'll get
+    // right to the beginning of the previous word
+    while (index > 0 && !isWordChar(this.text[index])) {
+        index --;
+    }
+
+    // while the current character is a word one, we skip it until we
+    // find the beginning of the current word
+    while (index > 0 && isWordChar(this.text[index - 1])) {
+        index --;
+    }
+
+    return index;
+};
+
+StringMorph.prototype.nextWordFrom = function (aSlot) {
+    var index = aSlot;
+    
+    while (index < this.endOfLine() && !isWordChar(this.text[index])) {
+        index ++;
+    }
+
+    while (index < this.endOfLine() && isWordChar(this.text[index])) {
+        index ++;
```

Comment:
haha, yeah no problem, let me change this. Too many languages in my head lately...